lib/suse/connect/api.rb

```diff
@@ -16,7 +16,15 @@ def initialize(response)
         end
       end
 
-      CONNECT_API_URL = 'https://scc.suse.com/connect'.freeze
+      def connect_api
+        uri_string = Settings.try(:scc).try(:host) || 'https://scc.suse.com/connect'
+        unless URI::DEFAULT_PARSER.make_regexp(['http', 'https', 'localhost']).match?(uri_string)
```

Comment:
So I managed to come around this check by using this string: `htxxtp://xxxxxxlocalhost:3000` (I swear this is the first try to trigger the exception!). I wonder, do we really need to validate the URI for sanity here?

If you run it with a malformed URI like the above you get:

```
SCC API request failed. Error details:
Request URL: htxxtp://xxxxxxlocalhost:3000/organizations/products?page=1
Response code: 0
Response body:
```

Isn't this enough for the user to find out the URI is just malformed? I'm undecided but would drop the check do make the code more easy and remove complexity. What do you think?

Reply:
We need to have the sanity checks. It's a cleaner way of handling the URIs. That url should've failed, I will check why it didn't.

Reply:
Turns out the `localhost` regexp matcher will accept any string, including the malformed one you just caught. I have removed it and tests are now passing.